test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,756 @@
+				unhealthy, err = oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", operatorNamespace, "subscription", ossmSubscriptionName, `-o=jsonpath={.status.conditions[?(@.type=="CatalogSourcesUnhealthy")].status}`).Output()
+				if err != nil {
+					e2e.Logf("Failed to check %s, error: %v, try next round", ossmSubscriptionName, err)
+				}
+				e2e.Logf("wait CatalogSourcesUnhealthy status to be False, and got %s", unhealthy)
+				return unhealthy
+			}, 5*time.Minute, time.Second).Should(o.Equal("False"))
+
+			g.By(fmt.Sprintf("wait untill the the OSSM csv %s is automatically created successfully", csvName))
+			var phase string
+			o.Eventually(func() string {
+				var err error
+				phase, err = oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", operatorNamespace, "csv", csvName, `-o=jsonpath={.status.phase}`).Output()
+				if err != nil {
+					e2e.Logf("Failed to check %s, error: %v, try next round", csvName, err)
+				}
+				e2e.Logf(fmt.Sprintf(" wait phase to be Succeeded, and got %s", phase))
+				return phase
+			}, 5*time.Minute, time.Second).Should(o.Equal("Succeeded"))
+
+			// deleted the istiod deployment and then checked if it was restored
+			deleteDeploymentAndWaitAvailableAgain(oc, istiodDeployment, ingressNamespace)
+
+			// deleted the istio and check if it was restored
+			g.By(fmt.Sprintf("try to delete the istio %s", istioName))
+			output, err := oc.AsAdmin().WithoutNamespace().Run("delete").Args("-n", ingressNamespace, "istio/"+istioName).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(output).To(o.ContainSubstring("deleted"))
+
+			g.By(fmt.Sprintf("wait untill the the istiod %s is automatically created successfully", istioName))
+			o.Eventually(func() string {
+				readyReplicas, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", ingressNamespace, "istio/"+istioName, `-o=jsonpath={.status.revisions.ready}`).Output()
+				if err != nil {
+					e2e.Logf("Failed to check istio %s, error: %v, try next round", istioName, err)
+				}
+				e2e.Logf("wait the ready replicas to be 1, and got %s", readyReplicas)
+				return readyReplicas
+			}, 5*time.Minute, time.Second).Should(o.Equal("1"))
+		})
+
+		g.It("and ensure gateway loadbalancer service and gateway dnsrecords get recreated after deleting them", func() {
+			const (
+				operatorNamespace = "openshift-operators"
+				ingressNamespace  = "openshift-ingress"
+				gatewayDeployment = "gateway-openshift-default"
+				gatewayLbService  = "gateway-openshift-default"
+				gatewayName       = "gateway"
+			)
+
+			// ensure default gateway objects is created
+			coreClient := clientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Getting the default domain")
+			defaultIngressDomain, err := getDefaultIngressClusterDomainName(oc, time.Minute)
+			o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
+			defaultDomain := strings.Split(defaultIngressDomain, "apps.")[1]
+
+			g.By("create the default API Gateway")
```

Comment:
It is not `API Gateway`, maybe you mean the default/custom gateway ? or maybe we don't need so many `g.By` ?

Reply:
copied from melvin's codes, wait for his update, will copy the codes after the merged.

Reply:
cc @melvinjoseph86

Reply:
i will rectify the same